mistralrs-core/src/pipeline/normal.rs

```diff
@@ -228,10 +228,9 @@ impl Loader for NormalLoader {
             Device::Cpu
         };
 
-        let mut is_lora = false;
+        let is_lora = self.kind.is_adapted_and(|a| a.is_lora());
```

Comment:
The PR enables setting `is_lora` prior to the match arms which simplifies that a bit 👍 

---

Not quite sure what you envisioned for the API here.

I decided to mirror `is_some` + `is_some_and` that exist for `Option` type, this way you have a bit more flexibility and under the hood it handles the vec iteration without having to think about it.

**NOTE:** I have not applied this to `gguf.rs` and `ggml.rs` as I have a large refactoring PR already touching that. I can apply the same changes there after this PR is merged.

Reply:
> I decided to mirror is_some + is_some_and that exist for Option type, this way you have a bit more flexibility and under the hood it handles the vec iteration without having to think about it.

I like this API. The old way with the boolean wasn't very idiomatic anyway.

> NOTE: I have not applied this to gguf.rs and ggml.rs as I have a large refactoring PR already touching that. I can apply the same changes there after this PR is merged.

Sounds like a good idea.